### PR TITLE
Added test for indexing a glob.

### PR DESCRIPTION
**Pull request by kshakir**

### CODE REVIEW

**Comment by mcovarr** [comment]
:+1:

[Approved with PullApprove](https://pullapprove.com/broadinstitute/centaur/pull-request/162/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/centaur)

File: src/main/resources/standardTestCases/globbingindex/globbingindex.wdl [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+task read_task {
+    Array[File] staticFiles
+    Array[File] staticRef
+    File staticFile
+    Array[File] staticArray
+    File globFile
+    Array[File] globArray
+
+    command { : }
```
Comment (cjllanwarne) [inline src/main/resources/standardTestCases/globbingindex/globbingindex.wdl]: `:`?

File: src/main/resources/standardTestCases/globbingindex/globbingindex.wdl [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    output {
+        String staticFilesContent = read_string(staticFiles[0])
+        String staticRefContent = read_string(staticRef[0])
+        String staticFileContent = read_string(staticFile)
+        String staticArrayContent = read_string(staticArray[0])
+        String globFileContent = read_string(globFile)
+        String globArrayContent = read_string(globArray[0])
```
Comment (cjllanwarne) [inline src/main/resources/standardTestCases/globbingindex/globbingindex.wdl:22]: ToL: why not include these outputs in the `glob_task` output block?

File: src/main/resources/standardTestCases/globbingindex/globbingindex.wdl [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        echo "globArray" > globArray.txt
+    }
+
+    output {
+        Array[File] staticFiles = ["staticFile.txt"]
+        Array[File] staticRef = staticFiles
+        File staticFile = staticFiles[0]
+        Array[File] staticArray = ["staticArray.txt"]
+        File globFile = glob("globFile.txt")[0]
+        Array[File] globArray = glob("globArray.txt")
```
Comment (cjllanwarne) [inline src/main/resources/standardTestCases/globbingindex/globbingindex.wdl:15]: ToL: Bonus points for `File globFile2 = glob("globFile.txt")[read_int("intFile.txt")]`...

**Review by cjllanwarne — APPROVED** [review]
The tests look good. I'm curious whether there are any older tests that are now redundant and could be wholesale replaced by this new one?

File: src/main/resources/standardTestCases/globbingindex/globbingindex.wdl (hunk 3 above)
Comment (mcovarr) [reply src/main/resources/standardTestCases/globbingindex/globbingindex.wdl:15]: Yeah I think this is the only thing that would test both `lhs` and `rhs`

**Comment by mcovarr** [comment]
:+1: with the test Chris suggests

File: src/main/resources/standardTestCases/globbingindex.test [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+metadata {
+  workflowName: globbingindex
+  status: Succeeded
+  "outputs.globbingindex.globFileContent": "globFile"
+  "outputs.globbingindex.globArrayContent": "globArray"
+  "outputs.globbingindex.staticFileContent": "staticFile"
+  "outputs.globbingindex.staticFilesContent": "staticFile"
+  "outputs.globbingindex.staticRefContent": "staticFile"
+  "outputs.globbingindex.staticArrayContent": "staticArray"
```
Comment (cjllanwarne) [inline src/main/resources/standardTestCases/globbingindex.test:16]: Depending on your OCD level, you might also want to add some `globLength = length(globArray)` outputs and verify that the globs are the correct length here.

**Comment by cjllanwarne** [comment]
👍. If the extra test works out of the box great... if not I'd add it as a new bug ticket rather than delaying this PR for it

[Approved with PullApprove](https://pullapprove.com/broadinstitute/centaur/pull-request/162/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/centaur)

File: src/main/resources/standardTestCases/globbingindex/globbingindex.wdl (hunk 1 above)
Comment (kshakir) [reply src/main/resources/standardTestCases/globbingindex/globbingindex.wdl]: http://stackoverflow.com/questions/3224878/what-is-the-purpose-of-the-colon-gnu-bash-builtin/3224910#3224910

File: src/main/resources/standardTestCases/globbingindex/globbingindex.wdl (hunk 3 above)
Comment (kshakir) [reply src/main/resources/standardTestCases/globbingindex/globbingindex.wdl:15]: There are a number of expressions that "look good to me?"™ but either don't parse `WdlExpression.fromString(<expr>)`, or return an AST that is effectively empty. I'll add them to a new ticket.

For example:
- `myArray[read_int("intFile.txt")]`
- `[0, 1, 2][0]`
- `{ "a": 1 }[a]`

Comment (kshakir) [reply src/main/resources/standardTestCases/globbingindex/globbingindex.wdl:15]: https://github.com/broadinstitute/wdl4s/issues/90

File: src/main/resources/standardTestCases/globbingindex.test (hunk 4 above)
Comment (kshakir) [reply src/main/resources/standardTestCases/globbingindex.test:16]: Done.

**Pull request merged**